### PR TITLE
[ENH]: Remove decorator from glm/regression

### LINKED ISSUE #3903: dead code: decorator for deprecated arguments
The decorator introduced in glm.regression in this commit seems now unused.

https://github.com/nilearn/nilearn/commit/e6591d8907ceae55eea6f04ce181acb6a12743b5

I wonder if it is better to remove it or to move it to _utils.helpers where other deprecation handling tools are located and to do some refactoring where possible to "standardize" how deprecation warnings are sent.

### PR DESCRIPTION
**Pull request by achamma723**
Closes #3903

### CODE REVIEW

**Comment by ymzayek** [comment]
Thx @achamma723 ! I think minor unrelated changes went into first_level.py and test_first_level.py. It's not a big deal since it's very minimal but if you could remove those that would be great!

File: doc/changes/latest.rst [hunk 1]
```diff
@@ -60,4 +60,5 @@ Changes
 - Refactor error raising tests using context managers (:gh:`3854` BY `François Paugam`_)
 - Added warning to deprecate `darkness` in ``surf_plotting._compute_facecolors_matplotlib`` and ``html_surface._get_vertexcolor`` (:gh`3855` by `Alisha Kodibagkar`_)
 - :bdg-secondary:`Doc` Replace skipped doctests with default code-blocks (:gh:`3681` in by `Patrick Sadil`_)
-- Move the `~nilearn.plotting.html_surface._mix_colormaps` to `cm.py` in :mod:`~nilearn.plotting` (:gh:`3919` by `Ahmad Chamma`_)
\ No newline at end of file
+- Remove deprecation decorator from `~nilearn.glm.regression` (:gh:`3921` by `Ahmad Chamma`_)
+- Move the `~nilearn.plotting.html_surface._mix_colormaps` to `cm.py` in :mod:`~nilearn.plotting` (:gh:`3919` by `Ahmad Chamma`_)
```
Comment (ymzayek) [inline doc/changes/latest.rst:64]: Hm the entry from the last PR shouldn't be showing up as a change here

Comment (achamma723) [reply doc/changes/latest.rst]: @ymzayek True, I think it was a merge conflict. Is it good now?

**Review by bthirion — APPROVED** [review]
Thx !

**Review by ymzayek — APPROVED** [review]
Thx @achamma723

**Pull request merged**